### ISSUE TITLE
[no-dupe-args][no-redeclare] False positive with object destructuring and inline type

### ISSUE THREAD

**Issue by armano2** [bradzacher/eslint-plugin-typescript#164]
**Tell us about your environment**	
* **eslint-plugin-typescript:**	20.1.1

**Please show your full configuration:**	
```js
module.exports = {
  root: true,
  env: {
    browser: false,
    es6: true,
    node: false
  },
  extends: [
    'plugin:vue/essential',
    '@vue/standard'
  ],
  plugins: ['typescript'],
  parserOptions: {
    parser: 'typescript-eslint-parser',
    jsx: false
  }
}
```
**What did you do? Please include the actual source code causing the issue.**	
```ts
function foo ({ v1, v2 }: { v1: number, v2: number }) {
}
```
**What did you expect to happen?**
year and month is not duplicated, its inline type definition

**What actually happened? Please include the actual, raw output from ESLint.**
```
Duplicate param 'v1'      no-dupe-args
Duplicate param 'v2'     no-dupe-args
'v1' is already defined   no-redeclare
'v2' is already defined  no-redeclare
```

**Comment by armano2** [comment]
issue is with parser not this plugin 👍 
visitor-tree is not emitting it as type
